python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -120,7 +142,7 @@ def __init__(
         )
 
 
-class Write(AbstractMap):
+class Write(AbstractUDFMap):
```

Comment:
This will be done in https://github.com/ray-project/ray/pull/32538. This PR is just refactoring whatever it's written.

Maybe we can also change Read to extend AbstractMap, but Read is not mapping any input op so it's not exactly matching AbstractMap.

Reply:
Wait, in this case, I don't see a strong benefit to have a separate `AbstractMap` now. Looks like `Write` is the only subclass of `AbstractMap`. Can we just extend `LogicalOperator`? cc @clarkzinzow.

Reply:
I think there is still benefit: the clarity in modeling Write and more accurate modeling of all those UDF mapping.

Reply:
Got it, if `AbstractMap` naming is confusing, we have another option:
* Rename `AbstractMap` to `AbstractUDFMap`
* Make `Write` extend `LogicalOperator`.

Honestly I don't see a major benefit to better model `Write` with the new `AbstractMap` here. Maybe I am missing something?

Reply:
It's not just naming, but also they are the same at the abstract level (after taking the UDF out etc).

Reply:
Actually, it should be fine to model Read as AbstractMap as well even though it has no input op. It doesn't look too bad to make the input op to Abstract as optional, unless there is concern from you guys.

Reply:
I am ok with `AbstractMap(input_op: Optional[...])`, and `Read` extends `AbstractMap` as well.

Reply:
Yep that sounds good to me too!